docker/run.bash

```diff
@@ -18,17 +18,8 @@ ARGS=("$@")
 # This is necessary so Gazebo can create a context for OpenGL rendering
 # (even headless).
 XAUTH=/tmp/.docker.xauth
-if [ ! -f $XAUTH ]
```

Comment:
Was this change necessary?

Reply:
Yes, previous code not working, here is example on a system with empty `/tmp`:
```
xauth:  /tmp/.docker.xauth not writable, changes will be ignored
xauth: (argv):1:  unable to read any entries from file "(stdin)"
chmod: changing permissions of '/tmp/.docker.xauth': Operation not permitted
Authorization required, but no authorization protocol specified
```

and previous code somehow creates `/tmp/.docker.xauth` as a directory

Reply:
It still quite simple to verify. But results can be false positive (it will be mentioned above error, but gui will work) if  `xhost +local` was executed before (not sure, maybe it should be local:root, it's still part of ancient magic)

Reply:
I just tried this locally and it didn't work for me. Running
```
XAUTH=/tmp/.docker.xauth
xauth nlist $DISPLAY | sed -e 's/^..../ffff/' | xauth -f $XAUTH nmerge -
```
I get 
```
xauth:  file /tmp/.docker.xauth does not exist
```

Reply:
I double checked and both (old and new) implementations are working fine. AFAIR old version initially didn't work. Maybe xhost +x has some permanent effect(still applied after reboot)
Tried your experiment, just run a command outside of the script and got the same error message. 

TBH, it's not my area of expertise, so I'll rollback it with minor change from :0 to $DISPLAY